src/credentials/DeviceAttestationConstructor.h

```diff
@@ -0,0 +1,65 @@
+/*
+ *
+ *    Copyright (c) 2021 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+#pragma once
+
+#include <lib/core/CHIPError.h>
+#include <lib/support/Span.h>
+
+#include <vector>
+
+namespace chip {
+namespace Credentials {
+
+/**
+ *  @brief Take the attestation elements buffer and return each component seperately.
+ *
+ *  @param[in]    attestationElements Buffer containg source of attestion
+ *  @param[out]   certificationDeclaration
+ *  @param[out]   attestationNonce
+ *  @param[out]   timestamp
+ *  @param[out]   firmwareInfo
+ *  @param[out]   vendorReservedArray
+ *  @param[inout] vendorReservedArraySize
+ *  @param[out]   vendorId (from vendor reserved elements)
+ *  @param[out]   profileNum (from vendor reserved elements)
```

Comment:
What happens if any of the buffers are too for the bytespans? What does it look like for firmwareInfo to be missing? How can you tell? What are the possible errors, in which conditions? Please fill this to make it usable without reverse engineering the implementation.

Reply:
will update